core/io/marshalls.cpp

```diff
@@ -670,7 +685,15 @@ Error decode_variant(Variant &r_variant, const uint8_t *p_buffer, int p_len, int
 							(*r_len) += used;
 						}
 
-						obj->set(str, value);
+						if (str == "script" && value.get_type() == Variant::STRING) {
+							String path = value;
+							ERR_FAIL_COND_V_MSG(path.is_empty() || !path.is_resource_file(), ERR_INVALID_DATA, "Invalid script path: '" + path + "'.");
```

Comment:
Is the resource file a safety restriction here, i.e. arbitrary code execution? Otherwise, unless I'm mistaken, scripts can be loaded from anywhere

Reply:
There is a similar restriction in text serialization.

https://github.com/godotengine/godot/blob/107f2961ccfac179af7682eb5f6e7ea91e80040c/core/variant/variant_parser.cpp#L1989-L1991

I think `res://` can be considered a trusted source, unliker `user://`, `http://`, etc.

https://github.com/godotengine/godot/blob/107f2961ccfac179af7682eb5f6e7ea91e80040c/core/string/ustring.cpp#L4623-L4625

Reply:
Then it's fine I think, parity is good here, could look into future optional permission for it with a new argument, but I think matching the behaviour of the text version makes sense 🙂